Explorer/Assets/DCL/SDKComponents/AvatarAttach/Systems/AvatarAttachHandlerSystem.cs

```diff
@@ -0,0 +1,134 @@
+using Arch.Core;
+using Arch.System;
+using Arch.SystemGroups;
+using Arch.SystemGroups.Throttling;
+using DCL.AvatarRendering.AvatarShape.UnityInterface;
+using DCL.Character.Components;
+using DCL.Diagnostics;
+using DCL.ECSComponents;
+using DCL.SDKComponents.AvatarAttach.Components;
+using DCL.Utilities;
+using ECS.Abstract;
+using ECS.Groups;
+using ECS.LifeCycle;
+using ECS.LifeCycle.Components;
+using ECS.Unity.Transforms.Components;
+using UnityEngine;
+
+namespace DCL.SDKComponents.AvatarAttach.Systems
+{
+    [UpdateInGroup(typeof(SyncedPostRenderingSystemGroup))]
+    [LogCategory(ReportCategory.SDK_COMPONENT)]
+    [ThrottlingEnabled]
+    public partial class AvatarAttachHandlerSystem : BaseUnityLoopSystem, IFinalizeWorldSystem
+    {
+        private AvatarBase ownPlayerAvatarBase;
+
+        public AvatarAttachHandlerSystem(World world, WorldProxy globalWorld) : base(world)
+        {
+            globalWorld.Query(new QueryDescription().WithAll<PlayerComponent, AvatarBase>(), (ref PlayerComponent playerComponent, ref AvatarBase avatarBase) => { ownPlayerAvatarBase = avatarBase; });
+        }
+
+        protected override void Update(float t)
+        {
+            SetupAvatarAttachQuery(World);
+            UpdateAvatarAttachTransformQuery(World);
+            HandleComponentRemovalQuery(World);
+            HandleEntityDestructionQuery(World);
+        }
+
+        [Query]
+        [None(typeof(AvatarAttachComponent))]
+        private void SetupAvatarAttach(in Entity entity, ref TransformComponent transformComponent, ref PBAvatarAttach pbAvatarAttach)
+        {
+            var component = new AvatarAttachComponent();
+
+            component.anchorPointTransform = GetAnchorPointTransform(pbAvatarAttach.AnchorPointId);
```

Comment:
you probably can present a constructor with a parameter for this

Reply:
The thing is: that anchor point transform can only be get from the `AvatarBase` component on the main player entity that exists in the global world.

I already tried other approaches before, mainly trying to run the `globalWorld` query to get the `AvatarBase` of the main player at plugin-level or even before that, but due to the order in which the different plugins and systems are initialized, that's not currently possible.

1. `AvatarPlugin` is initialized [at DynamicWorldContainer](https://github.com/decentraland/unity-explorer/blob/9e304c3cccc04676f3dfa23b2ade65024a7c8f40/Explorer/Assets/Scripts/Global/Dynamic/DynamicWorldContainer.cs#L101) (global level)
2. `AvatarAttachHandlerSystem` system is initialized [at StaticContainer](https://github.com/decentraland/unity-explorer/blob/e75ab2724167f4f747b529eccd754e8013be1bac/Explorer/Assets/Scripts/Global/StaticContainer.cs#L166) (scene level)
3. The Player entity is initialized [at GlobalWorldFactory](https://github.com/decentraland/unity-explorer/blob/9e304c3cccc04676f3dfa23b2ade65024a7c8f40/Explorer/Assets/Scripts/Global/Dynamic/GlobalWorldFactory.cs#L115). At that point the `StaticContainer` already exists and already initialized the `AvatarAttachHandlerSystem` and that's why I can't hook the player entity avatar base into the initialization of the `AvatarAttachHandlerSystem` system.

There may be another way involving some refactor maybe

Reply:
I meant only to present constructor with parameter to pass argument. And tiny style thingy - public field should be PascalCase. So smth like this

```
public struct AvatarAttachComponent
    {
        public Transform AnchorPointTransform;

        public AvatarAttachComponent(Transform anchorPoint)
        {
            AnchorPointTransform = anchorPoint;
        }
    }
```
    
with  ` var component = new AvatarAttachComponent(GetAnchorPointTransform(pbAvatarAttach.AnchorPointId));` call. 

But this would also work fine (without presenting a constructor)

```
 var component = new AvatarAttachComponent
            {
                AnchorPointTransform = GetAnchorPointTransform(pbAvatarAttach.AnchorPointId)
            };
```

Reply:
Ah I see what you mean now, I like it, sure I'll update it!

Reply:
[Done](https://github.com/decentraland/unity-explorer/pull/267/commits/9bd7f7a34a608b4a336e39e256c4063a0d5feb4b)!